docs/more_details/integration_tests.md

```diff
@@ -0,0 +1,121 @@
+		Method:             http.MethodGet,
+		Url:                "/euro/10/GBP",
+		Headers:            map[string]string{},
+		ExpectedStatusCode: http.StatusOK,
+		Assert: func(response *resty.Response) error {
+			result, err := strconv.ParseFloat(string(response.Body()), 64)
+			s.NoError(err)
+			s.Equal(8.0, result)
+
+			return nil
+		},
+	}
+}
+```
+
+Notice the predefined structure of an _ApiServerTestCase_: it is an object that has fields for all the information needed in performing an HTTP call, the expected status code, and a method that tests the result for correctness.
+
+In the example from `examples/getting_started/integration/api_test.go` we see two types of test cases (_resty.Client_ and _ApiServerTestCase_) in the same test suite. In fact, all three types of tests can be part of the same test suite.
+
+### Stream integration tests
+
+If you have a Gosoline application that takes its input from a stream, you need a test which can run your application locally, send data to a stream, and check your application's output for correctness. 
+
+The stream-consumer application, found in more_details/stream-consumer, reads unsigned integers from the inputEvent input, increments them by one, and publishes them to the publisher-outputEvent output. Below is an extract from its integration test:
+
+[embedmd]:# (../../examples/more_details/stream-consumer-test/stream_consumer_test.go /func \(s \*ConsumerTestSuite\) SetupSuite/ /\n}/)
+```go
+func (s *ConsumerTestSuite) SetupSuite() []suite.Option {
+	return []suite.Option{
+		suite.WithLogLevel("debug"),
+		suite.WithConfigFile("../stream-consumer/config.dist.yml"),
+		suite.WithModule("consumer", stream.NewConsumer("consumer", consumer.NewConsumer())),
+	}
+}
+```
+
+It is making use of the same `config.dist.yml` file as `stream-consumer`, and it will be using the module created by _consumer.NewConsumer_. A _StreamTestCase_ is very similar to an _ApiServerTestCase_:
+
+[embedmd]:# (../../examples/more_details/stream-consumer-test/stream_consumer_test.go /func \(s \*ConsumerTestSuite\) TestSuccess/ /\n}/)
+```go
+func (s *ConsumerTestSuite) TestSuccess() *suite.StreamTestCase {
+	return &suite.StreamTestCase{
+		Input: map[string][]suite.StreamTestCaseInput{
+			"consumer": {
+				{
+					Attributes: nil,
+					Body:       mdl.Uint(5),
+				},
+			},
+		},
+		Assert: func() error {
+			var result int
+			s.Env().StreamOutput("publisher-outputEvent").Unmarshal(0, &result)
+
+			s.Equal(6, result)
+
+			return nil
+		},
+	}
+}
+```
+
+This _StreamTestCase_ is an object defining an input and an `Assert` function. Gosoline will run the module, use this `StreamTestCase.Input` as input for it, then run `Assert`. The `Assert` function reads the first element from a stream, _publisher-outputEvent_, then compare it with an expected result. Notice that the input has a key called `consumer`, because in this application's `config.dist.yml` file, we have configured an input named `consumer`.
```

Comment:
Can we rename the input to something else than `consumer`? Right now you have the consumer called `consumer` reading from an input called `consumer` which can easily get mixed up. Maybe we should keep the consumer called `consumer`, the input called `consumerInput` and keep the input reading from a queue called `inputEvent` (as it currently does)

Reply:
StreamTestCase.Input -> StreamTestCaseInput

Reply:
Also the input was renamed, I think we can therefore remove the notice about the input being called "consumer" (which is now wrong anyway)